src/server/frontend_wayland/wl_keyboard.cpp

```diff
@@ -93,20 +112,20 @@ void mf::WlKeyboard::key(std::chrono::milliseconds const& ms, WlSurface* surface
     {
         log_warning(
             "Sending key 0x%2.2x %s to wl_surface@%u even though it was not explicitly given keyboard focus",
-            scancode, down ? "press" : "release", wl_resource_get_id(surface->resource));
+            scancode, down ? "press" : "release", wl_resource_get_id(surface.resource));
 
         focussed(surface, true);
     }
 }
 
-void mf::WlKeyboard::focussed(WlSurface* surface, bool should_be_focused)
+void mf::WlKeyboard::focussed(WlSurface& surface, bool should_be_focused)
```

Comment:
Why this change of signature? It seems to lead to a lot of needless diff.

Reply:
I guess I didn't need to do that. Did it for consistency with the new code.